app/src/main/java/com/mpieterse/stride/ui/layout/central/viewmodels/HabitViewerViewModel.kt

```
@@ -0,0 +1,202 @@
+                    error = "Check-ins load failed: ${cr.code ?: ""} ${cr.message ?: ""}"
+                )
+                return@launch
+            }
+        }
+
+        val today = LocalDate.now()
+        val monthStart = today.withDayOfMonth(1)
+        val completedThisMonth = mine.mapNotNull { checkin ->
+            try {
+                java.time.LocalDate.parse(checkin.dayKey)
+            } catch (e: Exception) { null }
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)